### PR TITLE
feat(theme): migrate devious-diamonds to yaml

### PR DESCRIPTION
**Pull request by ehawman-rosenberg**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md]
- [x] The commit message follows the [conventional commits][cc] guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

### Description

Migrated the devious-diamonds theme to YAML.

Implemented the .Writable property in Path (and learned a ton about Go text templates to get the spaces to work so it doesn't look wonky).

Discovered that ending a segment with ` \b` (there's a leading space that isn't rendering correctly) will ensure that it gets rendered, while eliminating the risk of accidentally deleting a character you want.

Did a ton of sorting. Very cathartic. Would recommend. Being able to collapse your YAML and have each segment's Type key/val on top is incredible
image

I was on vacation, hence the silence. YAML documentation is still on my radar. 😄 

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by ehawman-rosenberg** [comment]
Whoa there might be a problem with this (I'm thinking encoding?)

Attempting to use [this gist link](https://gist.githubusercontent.com/ehawman-rosenberg/f8de8315a091dee6cb69ae018ac27414/raw/devious-diamonds.omp.yaml) as a remote config (currently identical to the one submitted here) causes some weird issues.

```
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
comments:58:13: invalid char literal
comments:59:13: invalid char literal
comments:71:19: invalid char escape
comments:71:19: invalid char escape
comments:71:19: invalid char escape
comments:71:19: invalid char escape
comments:71:19: invalid char literal
comments:58:13: invalid char literal
comments:59:13: invalid char literal
comments:71:19: invalid char escape
comments:71:19: invalid char escape
comments:71:19: invalid char escape
comments:71:19: invalid char escape
comments:71:19: invalid char literal
Oh My Posh Error:

…ror found in #1 byte of ...|#yaml-langu|..., bigger context ...|#yaml-language-server:$schema=https:blocks:-type:pr|...
```

**Comment by ehawman-rosenberg** [comment]
I've edited the gist and no longer have invalid char problems.

However, the "Oh My Posh Error" is still around

`oh-my-posh debug` yields:

```
ReadMapCB: expect { or n, but found #, error found in #1 byte of ...|#yaml-langu|..., bigger context ...|#yaml-language-server:$schema=https:blocks:-type:pr|...
```

Some Googling dug up #1806.

**Comment by ehawman-rosenberg** [comment]
Oh for crying out loud. I squashed those pesky little commits. Now everything is ugly :(

**Comment by ehawman-rosenberg** [comment]
@JanDeDobbeleer is the reason my commitlint check is failing because I'm using the exclamation point to indicate a breaking change?

**Comment by JanDeDobbeleer** [comment]
You have a starting capital in the subject. Needs to be lowercase.

**Comment by ehawman-rosenberg** [comment]
@JanDeDobbeleer Currently my two subjects are `feat(theme)!: migrate devious-diamonds to YAML` with an additional message of `BREAKING CHANGE: File extension change from "json" to "yaml" will necessitate changes in config flags` and `fix(theme): corrected a spacing error in path` with no additional message.

Is BREAKING CHANGE causing the problem? I was trying to follow the conventional commit instructions.

**Comment by JanDeDobbeleer** [comment]
@ehawman-rosenberg footer's lines must not be longer than 100 characters [footer-max-line-length]

**Comment by ehawman-rosenberg** [comment]
@JanDeDobbeleer [101 characters](https://i.giphy.com/media/9G3wg7lH5DpxC/giphy.webp)

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -57,7 +57,7 @@ RUN pwsh -Command Install-Module posh-git -Scope AllUsers -Force; \
 ENV PATH "$PATH:/home/${USERNAME}/bin"
 
 # Can be used to override the devcontainer prompt default theme:
-ENV POSH_THEME="/workspaces/oh-my-posh/themes/jandedobbeleer.omp.json"
```
Comment (JanDeDobbeleer) [inline .devcontainer/Dockerfile:60]: @ehawman-rosenberg I missed this, but it seems like this wasn't the intention, right?

File: themes/devious-diamonds.omp.yaml [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+          fetch_worktree_count: true
+        template: "{{ .UpstreamIcon }}{{ .HEAD }}{{ .BranchStatus }}{{ if .Working.Changed }}  {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }}  {{ .Staging.String }}{{ end }}{{ if gt .StashCount 0 }}  {{ .StashCount }}{{ end }}"
+  - type: prompt
+    alignment: left
+    newline: true
+    segments:
+      - type: text
+        style: plain
+        foreground: cyan
+        template: " ╚"
```
Comment (JanDeDobbeleer) [inline themes/devious-diamonds.omp.yaml:185]: also, these should be the actual `\u` notations

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (ehawman-rosenberg) [reply .devcontainer/Dockerfile:60]: Is that not what it was? I had changed it to `devious-diamonds.omp.yaml` and I thought I was putting it back.

File: themes/devious-diamonds.omp.yaml (hunk 2 above)
Comment (ehawman-rosenberg) [reply themes/devious-diamonds.omp.yaml:185]: Oh no. I used an online converter and it spit the encoded chars out. I guess I'll have to redo it manually? Bleh.

What is the reasoning behind using `\u` notation? It's certainly easier to visually edit when they appear as the intended symbols.

Comment (ehawman-rosenberg) [reply themes/devious-diamonds.omp.yaml:185]: @JanDeDobbeleer `oh-my-posh config export --output ~/.mytheme.omp.yaml` exports the encoded characters. If it was set up to detect characters in the [Private Use Areas](https://en.wikipedia.org/wiki/Private_Use_Areas) and change their encoding, then that would give us a tool to do this automatically. I know I could make this work in Python, but I'm not confident enough to mess with your Go code on this one.

https://stackoverflow.com/questions/29161300/how-can-i-get-the-unicode-value-of-a-character-in-go

**Pull request merged**